mlir/lib/Dialect/MIOpen/Transforms/AffineTransforms.cpp

```diff
@@ -72,7 +75,8 @@ AffineMap AffineTransforms::buildIndexAffineMap(miopen::TransformOp op) {
           auto destDim = destDimAttr.getValue()[j].cast<IntegerAttr>().getInt();
 
           auto expr = getAffineDimExpr(destDim, op.getContext()) + getAffineConstantExpr(-leftPad, op.getContext());
-          if (leftPad == 0 && rightPad != 0) {
+
+          if (leftPad == 0 && rightPad != 0 & !closeRightPadding) {
```

Comment:
instead of using `closeRightPadding`, let's remove this whole block. we need to come up with a more holistic approach for OOB checks in the right padding case, now that we have enabled index diff map in #226 .